### ISSUE TITLE
make notification not depend on ActiveSupport

### ISSUE THREAD

**Issue by yarmand** [yammer/circuitbox#24]

**Comment by yarmand** [comment]
Notification should also be instantiated to not have to check they are available on every call.

**Comment by nickcampbell18** [comment]
What would you propose instead? Injectable dependency? Some other library?

**Comment by keram** [comment]
At ruby hacknight I tried tackle with active support extraction. It is still wip https://github.com/keram/circuitbox_activesupport_notifier https://github.com/keram/circuitbox/tree/activesupport_notifier_extraction but feel free share your thoughts if this could be alternative way to go ?

**Comment by sideshowcoder** [comment]
Love it Ill take a look asap thanks so much!

**Comment by yarmand** [comment]
@keram I like your implementation a lot !
You should rename the default notifier to be NullNotifier.
The AS notifier should stay in circuitbox just to be convenient I think.

**Comment by sideshowcoder** [comment]
I think the best way to go would be to have it in circuitbox, but make it so you have to require it explicitly if you want the active support one. 

```ruby
require "circuitbox"
require "circuitbox/active_support_notifier"
```

Will use the `AS::Notifications` while

```ruby
require "circuitbox"
```

Gives you circuitbox without it, this is kind of the same behavior as the
`FaradayMiddleware` has. Then we can provide a `NullNotifier`, which circuitbox
defaults to if nothing is supplied and if `active_support_notifier` is not
required. What do you think?

**Comment by matthewshafer** [comment]
this was fixed in #99
